src/logger.js

```diff
@@ -1,8 +1,165 @@
-var logger
-try {
-  logger = require('electron-timber')
-} catch (err) {
-  // if electron not available, fall back to console
-  logger = console
+const winston = require('winston')
+const path = require('path')
+const DailyRotateFile = require('winston-daily-rotate-file')
+const util = require('util')
+
+const store = require('./store')
+const appVersion = require('../package.json').version
+
+const BUGSNAG_API_KEY = 'fcd92279c11ac971b4bd29b646ec4125'
+
+let Bugsnag
+
+class Logger {
+  constructor () {
+    this.winston = winston.createLogger()
+    this.configured = false
+    this._messageQueue = []
+    this._debug = store.get('debugging')
+  }
+
+  configure ({
+    userDataPath,
+    isDev,
+    label
+  }) {
+    this.isDev = isDev
+    this._startBugsnag(isDev ? 'development' : 'production')
+    const prettyPrint = winston.format.printf(({ level, message, label, timestamp }) => {
+      return `${timestamp} [${label}] ${level}: ${message}`
+    })
+
+    this.winston.format = winston.format.combine(
+      winston.format.label({ label }),
+      winston.format.timestamp(),
+      prettyPrint
+    )
+    this.dirname = path.join(userDataPath, 'logs')
+
+    for (const transport of this.winston.transports) {
+      if (transport.close) {
+        transport.close()
+      }
+    }
+
+    const mainLog = new (DailyRotateFile)({
+      filename: '%DATE%.log',
+      dirname: this.dirname,
+      datePattern: 'YYYY-MM-DD',
+      zippedArchive: true,
+      maxFiles: '365d',
+      level: 'info'
+    })
+    mainLog.name = 'main'
+    mainLog.on('new', () => {
+      // if debugging is on and a new log file is created, turn it off
+      this.debugging(false)
+    })
+    mainLog.on('rotate', () => {
+      // if debugging is on during a rotate, turn it off
+      this.debugging(false)
+    })
+    this.winston.add(mainLog)
+
+    const errorTransport = new (DailyRotateFile)({
```

Comment:
Does this mean that errors are still going to a separate file? (I'm not familiar with how this module works.)

Reply:
Both!

Reply:
this can be turned off if we don't want it by removing the errorTransport lines entirely. Going to merge now